### PR TITLE
mgr/dashboard: remove RGW default realm restriction in NFS exports form, in case of single-sites RGW

### PR DESCRIPTION
**Pull request by torchiaf**
- remove RGW realm_name compare with default realm_name
- remove error message

Fixes: https://tracker.ceph.com/issues/57912
Signed-off-by: Francesco Torchia <francesco.torchia@suse.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts [hunk 1]
```diff
@@ -265,20 +265,7 @@ export class NfsFormComponent extends CdForm implements OnInit {
...
-                    mergeMap((isDefaultRealm) => {
-                      if (!isDefaultRealm) {
-                        throw new Error('Selected realm is not the default.');
-                      }
-                      return of(true);
-                    })
-                  )
-            )
-          )
+        ? this.rgwSiteService.get('realms')
```
Comment (torchiaf) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts:268]: I am not sure if these two api calls are necessary here, unless an error in `realms`/`filesystem` fetching is considered blocking for the NFS export creation.

**Comment by torchiaf** [comment]
> @torchiaf Please confirm that now NFS modules can handle buckets that do not belong to the default realm even when RGW-multisite is configured.

Hi @alfonsomthd , thanks for the reply.
I actually collected the following info about the issue:

1. NFS exports can be created and used for non-default realms, with a single-site RGW, using `CLI`; The same cannot be done using the Dashboard.
2. Probably NFS modules cannot handle buckets that do not belong to the default realm, in case of multi-site, but I don't tested this case, so I can't corfirm yet.

**Comment by alfonsomthd** [comment]
> In the case of point 2 is confirmed, we might consider to check if the RGW is single-site and allow to create NFS exports for non-default realm. What do you think ?
@torchiaf yeah, if the operation still fails when RGW-multisite + non-default realm is selected, let's not allow that particular action (that is, only allow it for single site) in order to not introduce a regression.

**Comment by torchiaf** [comment]
> @torchiaf yeah, if the operation still fails when RGW-multisite + non-default realm is selected, let's not allow that particular action (that is, only allow it for single site) in order to not introduce a regression.

I would allow it for single site at the moment. We could verify and eventually remove the restriction for multi-sites later, in a new PR.
I am going to change the current PR title to highlight I want to remove the restriction only for single-sites.

Do you know how to check if the RGW is running single-site or multi-site , on the dashboard code ?
Thanks.

**Comment by alfonsomthd** [comment]
> Do you know how to check if the RGW is running single-site or multi-site , on the dashboard code ? Thanks.

The only check that comes to my mind now is [this](https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-daemon-list/rgw-daemon-list.component.ts#L68).

**Comment by torchiaf** [comment]
> The only check that comes to my mind now is [this one](https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-daemon-list/rgw-daemon-list.component.ts#L68).

Thanks @alfonsomthd 
Looks like that function simply return the list of realms, but i thinks is not enough to check if multi-site is configured. Multiple realms can be created even in a single-site configuration, please correct me if I am wrong.

I asked some info to my teammate ( thanks @trociny ) .
It should be possible to check the list of `zones` (each site should have his own `zone` and there should be a` master-zone`), by the following CLI command:

`radosgw-admin period get`
`radosgw-admin sync status`

Here is an example output from my local environment:
```
master:~ # radosgw-admin period get
{
    "id": "863a4127-a2e2-42a3-ae1e-b287467ebc37",
    "epoch": 1,
    "predecessor_uuid": "",
    "sync_status": [],
    "period_map": {
        "id": "863a4127-a2e2-42a3-ae1e-b287467ebc37",
        "zonegroups": [],
        "short_zone_ids": []
    },
    "master_zonegroup": "",
    "master_zone": "",
    "period_config": {
        "bucket_quota": {
            "enabled": false,
            "check_on_raw": false,
            "max_size": -1,
            "max_size_kb": 0,
            "max_objects": -1
        },
        "user_quota": {
            "enabled": false,
            "check_on_raw": false,
            "max_size": -1,
            "max_size_kb": 0,
            "max_objects": -1
        }
    },
    "realm_id": "b1d1048b-dd8c-45bc-b6aa-8d7932d99ff4",
    "realm_name": "pippo2",
    "realm_epoch": 1
}
```

Probably we could assume that in case of `master_zone` is empty or error is returned by` period get` api, then single-site is configured.

What do you think?
Do we have a Rest Api which provide the same results ?

**Comment by trociny** [comment]
Actually, do we really need this single-site only restriction? Users could still create an exports and then setup multi-site later and I believe nothing would stop them from doing this. May be we should just show a warning when a non-default realm is used, that it might not work for multi-site sync? I.e. when previously we had an error now we will have just a warning.

**Comment by torchiaf** [comment]
> Actually, do we really need this single-site only restriction? Users could still create an exports and then setup multi-site later and I believe nothing would stop them from doing this. May be we should just show a warning when a non-default realm is used, that it might not work for multi-site sync? I.e. when previously we had an error now we will have just a warning.

@trociny  thanks for repying. Agree, showing the warning message at the end could be the best solution. What do you think @alfonsomthd  ?

**Comment by alfonsomthd** [comment]
> > Actually, do we really need this single-site only restriction? Users could still create an exports and then setup multi-site later and I believe nothing would stop them from doing this. May be we should just show a warning when a non-default realm is used, that it might not work for multi-site sync? I.e. when previously we had an error now we will have just a warning.
> 
> @trociny thanks for repying. Agree, showing the warning message at the end could be the best solution. What do you think @alfonsomthd ?

I'm involved in another projects these days, but what I remember is that the operation was failing on the conditions mentioned above.
@torchiaf IMHO checking if the error still persists should be the 1st thing: if that is not the case anymore, go ahead and delete the restriction. Otherwise, l would keep the checking to prevent an error being thrown (regression introduced) that could cause a bug will be opened for this.
@epuertat @nizamial09 Please share with @torchiaf your thoughts and next action item here.


`
https://docs.ceph.com/en/latest/radosgw/multisite/#create-a-realm
`

**Pull request closed without merging**